packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -300,13 +303,21 @@ class TextSelectionOverlay {
 
   /// The object supplied to the [CompositedTransformTarget] that wraps the text
   /// field.
-  final LayerLink layerLink;
+  final LayerLink toolbarLayerLink;
+
+  /// The objects supplied to the [CompositedTransformTarget] that wraps the
+  /// selection handles.
+  final List<LayerLink> handleLayerLinks;
 
   // TODO(mpcomplete): what if the renderObject is removed or replaced, or
   // moves? Not sure what cases I need to handle, or how to handle them.
   /// The editable line in which the selected text is being displayed.
   final RenderEditable renderObject;
 
+  /// The object supplied to [_TextSelectionHandleOverlay] that determines the
```

Comment:
This is surprising, why does the textDirection determine the handle type?

Reply:
textdirection will change the order of handles
ltr:
start -> left handle,  end -> right handle
rtl:
start -> right handle, end -> left handle

Reply:
but I think i can avoid this parameter, I was thinking of getting rid of renderObject dependency from this class, but it seems I will still need it to calculate the handlesize